go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -200,6 +200,81 @@ func NewNetworkPolicy(policy *knet.NetworkPolicy) *networkPolicy {
 	return np
 }
 
+func (bnc *BaseNetworkController) syncNetworkPolicies(networkPolicies []interface{}) error {
+	expectedPolicies := make(map[string]map[string]bool)
+	for _, npInterface := range networkPolicies {
+		policy, ok := npInterface.(*knet.NetworkPolicy)
+		if !ok {
+			return fmt.Errorf("spurious object in syncNetworkPolicies: %v", npInterface)
+		}
+		if nsMap, ok := expectedPolicies[policy.Namespace]; ok {
+			nsMap[policy.Name] = true
+		} else {
+			expectedPolicies[policy.Namespace] = map[string]bool{
+				policy.Name: true,
+			}
+		}
+	}
+	err := bnc.syncNetworkPoliciesCommon(expectedPolicies)
+	if err != nil {
+		return err
+	}
+
+	if bnc.NetInfo.IsSecondary() {
+		return nil
+	}
+
+	// add default hairpin allow acl
+	err = bnc.addHairpinAllowACL()
```

Comment:
Why don't we need the hairpin allow ACLs for primary networks? Add comment.

Reply:
added comment about why hairpin allow ACLs disabled now for UDN. will revisit once we have cluster port group exists per network. currently this is not impacting unit tests and e2e's.

Reply:
https://issues.redhat.com/browse/SDN-5290